File: src/main/java/com/grandtech/model/Teacher.java
package com.grandtech.model;

import jakarta.persistence.Column;
import jakarta.persistence.DiscriminatorValue;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;

/**
 * Entity representing a teacher user account.
 *
 * <p>Stored in the {@code teachers} table, joined to the {@code users} table
 * via the shared primary key. Profile fields are nullable on creation and
 * filled in by a subsequent profile-update request.
 */
@Entity
@Table(name = "teachers")
@DiscriminatorValue("TEACHER")
public class Teacher extends User {

    /** Full name of the teacher. */
    @Column
    public String name;

    /** Contact email address of the teacher. */
    @Column
    public String email;

    /** Teachers Service Commission registration number (optional). */
    @Column(name = "tsc_number")
    public String tscNumber;

    /**
     * Finds a teacher by email address.
     *
     * @param email the email address to look up
     * @return the matching {@link Teacher}, or {@code null} if not found
     */
    public static Teacher findByEmail(final String email) {
        return find("email", email).firstResult();
    }

    /**
     * Returns the discriminator value identifying this user as a teacher.
     *
     * @return {@code "TEACHER"}
     */
    @Override
    public String getType() {
        return "TEACHER";
    }
}

File: src/main/kotlin/com/grandtech/model/Teacher.kt
package com.grandtech.model

/**
 * A teaching staff member registered by a school for timetable scheduling.
 *
 * Stored as `(:Teacher)` linked to the school via `HAS_TEACHER` and to
 * their assigned learning areas via `TEACHES`.
 *
 * @property id                Neo4j-generated UUID; null before persistence
 * @property name              full name of the teacher
 * @property email             contact email address
 * @property phone             optional phone number
 * @property tscNumber         Teachers Service Commission registration number (optional)
 * @property maxPeriodsPerWeek upper bound on weekly lessons; defaults to 23
 * @property maxPeriodsPerDay  upper bound on daily lessons; defaults to 6
 * @property subjectIds        subject IDs — supplied on create/update (1–2 entries); returned on all read responses
 */
data class Teacher(
    val id: String? = null,
    val name: String? = null,
    val email: String? = null,
    val phone: String? = null,
    val tscNumber: String? = null,
    val maxPeriodsPerWeek: Int? = null,
    val maxPeriodsPerDay: Int? = null,
    val subjectIds: List<String>? = null,
)